src/iqm/cirq_iqm/serialize.py

```diff
@@ -30,7 +160,35 @@ def serialize_circuit(circuit: Circuit) -> iqm_client.Circuit:
     Returns:
         data transfer object representing the circuit
     """
-    instructions = tuple(map(map_operation, circuit.all_operations()))
+    total_ops_list = []
+    cc_prx_support = False
+    for moment in circuit:
+        for op in moment:
+            if isinstance(op, ClassicallyControlledOperation):
+                cc_prx_support = True
+            total_ops_list.append(op)
+    instructions = list(map(map_operation, total_ops_list))
+
+    if cc_prx_support:
+        mkey_to_measurement = {}
+        used_keys = set()
+        for inst in instructions:
+            if inst.name == 'measure':
+                if inst.args['key'] in used_keys:
+                    raise OperationNotSupportedError('Cannot use the same key for multiple measurements')
+                mkey_to_measurement[inst.args['key']] = inst
+                used_keys.add(inst.args['key'])
+            elif inst.name == 'cc_prx':
+                feedback_key = inst.args['feedback_key']
+                measurement = mkey_to_measurement.get(feedback_key)
+                if measurement is None:
+                    raise OperationNotSupportedError(
+                        f'cc_prx has feedback_key {feedback_key}, but no measure operation with that key precedes it.'
+                    )
+                if len(measurement.qubits) != 1:
+                    raise OperationNotSupportedError('cc_prx must depend on the measurement result of a single qubit.')
+                inst.args['feedback_qubit'] = measurement.qubits[0]
```

Comment:
```suggestion
                inst.args['feedback_qubit'] = measurement.qubits[0]
                measurement['feedback_key'] = feedback_key
```
As per @smite 's earlier comment